### PR TITLE
Use rust-xdg for cache creation

### LINKED ISSUE #10: Evaluate using xdg-basedir crate
The [xdg-basedir](https://kiran-kp.github.io/xdg-basedir/xdg_basedir/index.html) crate could make the $XDG_CACHE lookup (usually `~/.cache`) more standards compliant.

A `get_dir` function could be conditionally-compiled for unix and non-unix platforms. On unix, the $XDG_CACHE lookup would be used, on Windows a directory in the user profile that doesn't contain a dot would be used.

### PR DESCRIPTION
**Pull request by SShrike**
Pretty small PR that closes #10.

I believe this PR also fixes a bug you might not of noticed, where if a user changed `$XDG_CACHE_HOME`, tealdeer would still create the cache under `~/.cache`. Correct me if I'm wrong though.

I've also gone through and removed the spaces like [you said](https://github.com/dbrgn/tealdeer/issues/10#issuecomment-211429317).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,17 +17,18 @@ logging = ["env_logger"]
...
-docopt = "^0.6"
 time = "^0.1"
+flate2 = "^0.2"
+docopt = "^0.6"
 walkdir = "^0.1"
-env_logger = { version = "^0.3", optional = true }
+ansi_term = "^0.7"
+rustc-serialize = "^0.3"
 clippy = {version = "~0.0.37", optional = true}
+env_logger = { version = "^0.3", optional = true }
```
Comment (dbrgn) [inline Cargo.toml]: Haha, ordering by length. Is that some convention?

File: src/cache.rs [hunk 2]
```diff
@@ -46,11 +46,12 @@ impl Cache {
             }
         };
 
-        // Otherwise, fall back to ~/.cache/tealdeer
-        let home_dir = try!(env::home_dir().ok_or(
-            CacheError("Could not determine home directory".into())
-        ));
-        Ok(home_dir.join(".cache").join("tealdeer"))
+        // Otherwise, fall back to $XDG_CACHE_HOME/tealdeer.
+        let xdg_dirs = match BaseDirectories::with_prefix("tealdeer") {
```
Comment (dbrgn) [inline src/cache.rs]: Maybe we could use `with_prefix(::NAME)``?

File: src/cache.rs [hunk 3]
```diff
@@ -36,6 +35,7 @@ impl Cache {
         // $TLDR_RS_CACHE_DIR env variable.
```
Comment (dbrgn) [inline src/cache.rs]: Oh, we should probably rename that too (in a separate PR)...

Comment (SShrike) [inline src/cache.rs]: That was the next thing on my list, renaming everything from `tldr` to `tealdeer`.

File: src/cache.rs [hunk 4]
```diff
@@ -46,11 +46,12 @@ impl Cache {
...
 
-        // Otherwise, fall back to ~/.cache/tealdeer
-        let home_dir = try!(env::home_dir().ok_or(
-            CacheError("Could not determine home directory".into())
-        ));
-        Ok(home_dir.join(".cache").join("tealdeer"))
+        // Otherwise, fall back to $XDG_CACHE_HOME/tealdeer.
+        let xdg_dirs = match BaseDirectories::with_prefix("tealdeer") {
+            Ok(dirs) => dirs,
+            Err(_) => return Err(CacheError("$HOME does not exist or is not defined.".into())),
```
Comment (dbrgn) [inline src/cache.rs]: The XDG cache dir could also be defined depending on the `XDG_CACHE_HOME` variable or on other fallbacks. Maybe make this more generic?

    return Err(CacheError("Could not determine XDG base directory.".into()))

File: src/cache.rs (hunk 3 above)
Comment (dbrgn) [inline src/cache.rs]: I thought I did that, but missed some places...

File: Cargo.toml (hunk 1 above)
Comment (SShrike) [inline Cargo.toml]: Didn't even realise I did that haha, it's personal preference really. I'm not really sure how most Rustaceans organise their `Cargo.toml`.

File: src/cache.rs (hunk 3 above)
Comment (SShrike) [inline src/cache.rs]: I intend to rename things like `$TLDR_RS_CACHE_DIR`, `TldrError`, etc.

Comment (dbrgn) [inline src/cache.rs]: :+1:

File: src/cache.rs (hunk 2 above)
Comment (SShrike) [inline src/cache.rs]: Good call.

**Comment by SShrike** [comment]
Obligatory :tada:

**Pull request merged**